### PR TITLE
SAK-44740 check for categoryId from the GB as the instructor can edit in both Samigo and GB

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java [hunk 1]
```diff
@@ -797,12 +788,17 @@ public boolean updateGB(PublishedAssessmentSettingsBean assessmentSettings, Publ
 			Integer scoringType = evaluation.getScoringType();
 			if (evaluation.getToGradeBook()!=null && 
 					evaluation.getToGradeBook().equals(EvaluationModelIfc.TO_DEFAULT_GRADEBOOK.toString())){
-				Long categoryId = null;
+
+				// Can't trust the old assessment category id because the instructor could have changed inside Gradebook directly!
+				Long externalCategoryId = gbsHelper.getExternalAssessmentCategoryId(GradebookFacade.getGradebookUId(), assessment.getPublishedAssessmentId().toString(), g);
+				long currentCategoryId =  externalCategoryId != null ? externalCategoryId : -1;
+				long newCategoryId = NumberUtils.toLong(assessmentSettings.getCategorySelected(), -1);
+				boolean isCategoryChanged = (currentCategoryId > 0 || newCategoryId > 0) && currentCategoryId != newCategoryId;
```
Comment (ern) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java:796]: this reads as though 0 could be a valid category should these be (currentCategoryId > -1 || newCategoryId > -1)?

Comment (ottenhoff) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java:796]: is not a valid categoryId

Comment (ottenhoff) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java:796]: 0 would not indicate a valid categoryId. It must be greater than zero to indicate a selected category

Comment (ern) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java:796]: ok cool

**Review by ern — APPROVED** [review]

**Pull request merged**